src/identity/components/GlobalHeader/GlobalHeaderDropdown/index.tsx

```diff
@@ -0,0 +1,201 @@
+  }
+
+  private renderMainMenuOptions = () => {
+    const {mainMenuOptions} = this.props
+    return (
+      <div>
+        <hr className="line-break" />
+        {mainMenuOptions.map(value => {
+          const iconEl = <Icon glyph={value.iconFont} className="button-icon" />
+          const textEl = <span>{value.name}</span>
+          return (
+            <Dropdown.HrefItem key={value.name} href={value.href}>
+              {iconEl}
+              {textEl}
+            </Dropdown.HrefItem>
+          )
+        })}
+      </div>
+    )
+  }
+
+  private renderTypeAheadMenu = () => {
+    const {typeAheadMenuOptions} = this.props
+    const {selectedItem} = this.state
+    const {
+      typeAheadInputPlaceholder,
+      typeAheadOnSelectOption,
+      dropdownMenuStyle,
+    } = this.props
+    return (
+      <GlobalHeaderTypeAheadMenu
+        typeAheadPlaceHolder={typeAheadInputPlaceholder}
+        typeAheadMenuOptions={typeAheadMenuOptions}
+        onSelectOption={typeAheadOnSelectOption}
+        style={dropdownMenuStyle}
+        defaultSelectedItem={selectedItem}
+      />
+    )
+  }
+
+  private renderMenu = () => {
+    const {
+      mainMenuHeaderText,
+      dropdownMenuTheme = DropdownMenuTheme.None,
+      dropdownMenuStyle,
+      typeAheadMenuOptions,
+    } = this.props
+    const {showTypeAheadMenu} = this.state
+
+    const textEl = <span>{mainMenuHeaderText}</span>
+    const iconEl = (
+      <Icon
+        glyph={showTypeAheadMenu ? IconFont.CaretLeft_New : IconFont.Switch_New}
+        className="button-icon"
+      />
+    )
+    return (
+      <Dropdown.Menu
+        // testID={menuTestID}
+        theme={dropdownMenuTheme}
+        style={dropdownMenuStyle}
+      >
+        {/* Multi-org UI tickets #4051 and #4047, when user only has 1 account or 1 org, switch button is disabled */}
+        {typeAheadMenuOptions.length > 1 && (
+          <Dropdown.Item onClick={() => this.toggleShowTypeAheadMenu()}>
+            <FlexBox
+              justifyContent={JustifyContent.SpaceBetween}
+              alignItems={AlignItems.Center}
+            >
+              <span>
+                {iconEl}
+                {textEl}
+              </span>
+              {showTypeAheadMenu === false && (
+                <Icon
+                  glyph={IconFont.CaretRight_New}
+                  className="button-icon"
+                  style={{marginRight: 0}}
+                />
+              )}
+            </FlexBox>
+          </Dropdown.Item>
+        )}
+        {showTypeAheadMenu
+          ? this.renderTypeAheadMenu()
+          : this.renderMainMenuOptions()}
+      </Dropdown.Menu>
+    )
+  }
+
+  render() {
+    const {id, style, className} = this.props
+    const dropdownProps = {id, style, className}
+
+    return (
+      <Dropdown
+        {...dropdownProps}
+        disableAutoFocus
+        button={this.dropdownButton}
+        menu={() => this.renderMenu()}
```

Comment:
Same comment as above re: if possible, avoiding an anonymous function in a prop

Reply:
✅